inference-engine/src/mkldnn_plugin/CMakeLists.txt

```diff
@@ -161,6 +161,11 @@ ie_add_plugin(NAME ${TARGET_NAME}
 
 set_ie_threading_interface_for(${TARGET_NAME})
 
+if(SELECTIVE_BUILD STREQUAL "ON")
+    # After disabling a block of code, some variables might be unused.
+    target_compile_options(${TARGET_NAME} PRIVATE -Wno-unused-variable)
+endif()
```

Comment:
maybe we can disable such warnings at all for selective build? It seems that such issues can frequently happen for other cases / components in future.

Reply:
Yes, I completely agree with you. I proposed this approach earlier but someone wasn't agree with it due to possible errors with unused variables.

Reply:
https://github.com/openvinotoolkit/openvino/pull/3702

Reply:
I frequently see _inference-engine/build/_deps/ext_protobuf-src/src/google/protobuf/any.pb.cc:94:13: error: \u2018dynamic_init_dummy_google_2fprotobuf_2fany_2eproto\u2019 defined but not used [-Werror=unused-variable]
 static bool dynamic_init_dummy_google_2fprotobuf_2fany_2eproto = []() { AddDescriptors_google_2fprotobuf_2fany_2eproto(); return true; }();_. when I build locally on develop machine. It's really block develop process sometimes.

Reply:
I believe that error generation for unused variables should be disabled for all 3rd party libraries, but the current PR only affects the selective build mode.

Reply:
3rd party libraries were fixed separately.